client/src/services/task.ts

```diff
@@ -42,4 +48,23 @@ export class TaskService {
     const result = await axios.post<{ data: Task }>(`/api/task/`, data);
     return result.data.data;
   }
+
+  async createCriteriaForCourseTask(taskId: number, data: CriteriaData[]) {
+    const result = await axios.post<{ criteria: CriteriaData[] }>(`/api/v2/task/${taskId}/criteria`, {
+      criteria: data,
+    });
+    return result.data;
+  }
+
+  async getCriteriaForCourseTask(taskId: number) {
+    const result = await axios.get<{ criteria: CriteriaData[] }>(`/api/v2/task/${taskId}/criteria`);
+    return result.data.criteria;
+  }
+
+  async updateCriteriaForCourseTask(taskId: number, data: CriteriaData[]) {
+    const result = await axios.patch<{ criteria: CriteriaData[] }>(`/api/v2/task/${taskId}/criteria`, {
+      criteria: data,
+    });
+    return result.data;
+  }
```

Comment:
Please run

```
1. cd nestjs
2. npm run openapi
```

And all added by you DTO, methods, etc. will be generated to client/api and you will not need add them here manually